app/src/main/java/org/simple/clinic/editpatient/deletepatient/DeleteReasonItem.kt

```diff
@@ -0,0 +1,59 @@
+package org.simple.clinic.editpatient.deletepatient
+
+import androidx.annotation.StringRes
+import androidx.recyclerview.widget.DiffUtil
+import io.reactivex.subjects.Subject
+import kotlinx.android.synthetic.main.list_delete_reason.*
+import org.simple.clinic.R
+import org.simple.clinic.widgets.ItemAdapter
+import org.simple.clinic.widgets.recyclerview.ViewHolderX
+
+data class DeleteReasonItem(
+    @StringRes val displayText: Int,
+    val reason: PatientDeleteReason,
+    val isSelected: Boolean
+) : ItemAdapter.Item<DeleteReasonItem.Event> {
+
+  companion object {
+
+    fun from(
+        deleteReasons: List<PatientDeleteReason>,
+        selectedReason: PatientDeleteReason?
+    ): List<DeleteReasonItem> {
+      return deleteReasons.map { deleteReason ->
+        val displayText = when (deleteReason) {
+          PatientDeleteReason.Duplicate -> R.string.deletereason_duplicate_patient
+          PatientDeleteReason.AccidentalRegistration -> R.string.deletereason_accidental_registration
+          PatientDeleteReason.Died -> R.string.deletereason_died
```

Comment:
Are we doing this in multiple places? Might be better to have a `displayResId` property on the `PatientDeleteReason` type itself.

Reply:
We are using `PatientDeleteReason` in update and view renderer tests, so I wasn't sure having an Android string res there. WDYT?

Reply:
As long as the code path to query those are not executed in the tests, it should be fine.

Reply:
This will be similar to the labels we render for blood sugar types, where we do something similar, I think.

Reply:
Yeah, that works. Changed it to an enum class (We can continue using the `sealed class` as well but readability was this was cleaner)

```kotlin
enum class PatientDeleteReason(@StringRes val displayText: Int) {
  Duplicate(R.string.deletereason_duplicate_patient),
  AccidentalRegistration(R.string.deletereason_accidental_registration),
  Died(R.string.deletereason_died)
}
```